sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/Transforms.java

```diff
@@ -563,29 +558,40 @@ private static FormPage getFormPage(ReadResult readResultItem, List<FormTable> p
         }
         List<FormElement> formElementList = new ArrayList<>();
         elements.forEach(elementString -> {
-            String[] indices = NON_DIGIT_PATTERN.matcher(elementString).replaceAll(" ").trim().split(" ");
-
-            if (indices.length < 2) {
-                throw LOGGER.logExceptionAsError(new RuntimeException("Cannot find corresponding reference elements "
-                    + "for the field value."));
-            }
-
-            int readResultIndex = Integer.parseInt(indices[0]);
-            int lineIndex = Integer.parseInt(indices[1]);
-
-            if (indices.length == 3) {
-                int wordIndex = Integer.parseInt(indices[2]);
+            Matcher wordMatcher = Pattern.compile(WORD_REGEX).matcher(elementString);
+            Matcher lineMatcher = Pattern.compile(LINE_REGEX).matcher(elementString);
+            Matcher selectionMarkMatcher = Pattern.compile(SELECTION_MARK_REGEX).matcher(elementString);
```

Comment:
We should make these `Pattern`s static, compiling a pattern is expensive and this looks like a hot path